diff --git a/dist/RosFollower.js b/dist/RosFollower.js
index 72d7c6376ac6f8b3e07178bb672f89e78f52c303..51cbbecdebafa2ebefdc4c2f04e014a73ca6fbd4 100644
--- a/dist/RosFollower.js
+++ b/dist/RosFollower.js
@@ -1,6 +1,7 @@
 "use strict";
 Object.defineProperty(exports, "__esModule", { value: true });
 exports.RosFollower = void 0;
+const ipaddr = require("ipaddr.js");
 const xmlrpc_1 = require("@foxglove/xmlrpc");
 const eventemitter3_1 = require("eventemitter3");
 function CheckArguments(args, expected) {
@@ -107,7 +108,7 @@ class RosFollower extends eventemitter3_1.EventEmitter {
             this.emit("publisherUpdate", topic, publishers, callerId);
             return [1, "", 0];
         };
-        this.requestTopic = async (_, args) => {
+        this.requestTopic = async (_, args, req) => {
             const err = CheckArguments(args, ["string", "string", "*"]);
             if (err != null) {
                 throw err;
@@ -124,7 +125,9 @@ class RosFollower extends eventemitter3_1.EventEmitter {
             if (addr == undefined) {
                 return [0, "cannot receive incoming connections", []];
             }
-            const tcp = ["TCPROS", addr.address, addr.port];
+
+            const ip = ipaddr.process(req.socket?.localAddress ?? addr.address);
+            const tcp = ["TCPROS", ip.toString(), addr.port];
             return [1, "", tcp];
         };
         this._rosNode = rosNode;
